Model/Button/GetCartButtonConfig.php

```diff
@@ -119,7 +119,8 @@ public function getDiscountAmount($quote, AmwalButtonConfigInterface $buttonConf
             } else {
                 foreach ($quote->getAllVisibleItems() as $item) {
```

Comment:
I'm thinking, couldn't we use $quote->getTotals() and then do a foreach to retrieve the discount amount directly?

Something like:
```
$quoteTotals = $quote->getTotals();
foreach ($quoteTotals as $quoteTotal) {
    $discountAmount += $quoteTotal->getDiscountAmount();
}
```

Might be worth a try to simplify the calculations, but not sure if it will work in this case.

Reply:
I didn't try this, let me try.